### ISSUE TITLE
Fail with parametric type

### ISSUE THREAD

**Issue by iblis17** [JuliaDebug/ASTInterpreter2.jl#17]
```julia
julia> struct B{T} end

julia> (::B)() = 42

julia> @enter B{Int}()()
ERROR: type UnionAll has no field name
Stacktrace:
 [1] maybe_step_through_wrapper!(::Array{ASTInterpreter2.JuliaStackFrame,1}) at /home/iblis/.julia/v0.6/ASTInterpreter2/src/ASTInterpreter2.jl:422
```

**Comment by jamii** [comment]
@iblis17 

I'm not sure what the intended behavior is here, but a temporary workaround is to change the line to:

``` julia
parameter = Base.unwrap_unionall(stack[1].meth.sig).parameters[1]
is_kw = (:name in fieldnames(parameter)) && startswith(String(parameter.name.name), "#kw")
```

That at least lets the debugger continue, and it doesn't seem to have any other problems with this kind of code.

**Comment by staticfloat** [comment]
Just a note that this still happens and causes problems in `Flux` with BatchNorm, e.g.:

```
using Flux, ASTInterpreter2
x = randn(10, 10, 4, 1)
BN = BatchNorm(4)
```

Trying to `@enter BN(x)` gives:
```
julia> @enter BN(x)
ERROR: type UnionAll has no field name
Stacktrace:
 [1] getproperty(::Type, ::Symbol) at ./sysimg.jl:15
 [2] maybe_step_through_wrapper!(::Array{ASTInterpreter2.JuliaStackFrame,1}) at /home/staticfloat/.julia/dev/ASTInterpreter2/src/ASTInterpreter2.jl:392
 [3] top-level scope at /home/staticfloat/.julia/dev/ASTInterpreter2/src/ASTInterpreter2.jl:465
```

**Comment by KristofferC** [comment]
Still happens with #37

**Comment by timholy** [comment]
Fixed by #40
